### PR TITLE
Add Image and category support

### PR DESCRIPTION
**Pull request by pdalfarr**
Add Image and category support

### CODE REVIEW

File: .idea/modules.xml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <project version="4">
```
Comment (aritraroy) [inline .idea/modules.xml:2]: The .idea files should not be commited.

File: .idea/vcs.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (aritraroy) [inline .idea/vcs.xml:1]: Please remove these files from your commit.

File: app/src/main/java/com/aritraroy/rxmagnetodemo/MainActivity.java [hunk 3]
```diff
@@ -316,15 +326,63 @@ private void getRecentChangelog(String packageName) {
                 });
     }
 
-    private void showResult(String title, String result) {
+   private void getAppCategory(String packageName) {
+      final MaterialDialog progressDialog = showLoading(this, getResources().getString(R.string.message_grabbing));
+
+      Observable<Pair<String, String>> changelogObservable = rxMagneto.grabAppCategory(packageName);
```
Comment (aritraroy) [inline app/src/main/java/com/aritraroy/rxmagnetodemo/MainActivity.java:332]: Why is it named "changelogObservable"? It should be "appCategoryObservable".

File: app/src/main/java/com/aritraroy/rxmagnetodemo/MainActivity.java [hunk 4]
```diff
@@ -316,15 +326,63 @@ private void getRecentChangelog(String packageName) {
...
+                  progressDialog.dismiss();
+               }
+               showResult(getResources().getString(R.string.label_error), throwable.getMessage());
+            });
+   }
+
+   private void getAppImage(String packageName) {
+      final MaterialDialog progressDialog = showLoading(this, getResources().getString(R.string.message_grabbing));
+
+      Observable<String> changelogObservable = rxMagneto.grabAppImage(packageName);
```
Comment (aritraroy) [inline app/src/main/java/com/aritraroy/rxmagnetodemo/MainActivity.java:351]: Why is it named "changelogObservable"? It should be named "appImageObservable".

File: rxmagneto/src/main/java/com/aritraroy/rxmagneto/utils/Constants.java [hunk 5]
```diff
@@ -1,10 +0,0 @@
-package com.aritraroy.rxmagneto.utils;
```
Comment (aritraroy) [inline rxmagneto/src/main/java/com/aritraroy/rxmagneto/utils/Constants.java:1]: Why are these files deleted? You shouldn't remove any already available functionality from the library.

File: rxmagneto/src/main/java/com/aritraroy/rxmagneto/RxMagnetoInternal.java [hunk 6]
```diff
@@ -214,4 +225,83 @@ public void call(Emitter<ArrayList<String>> emitter) {
...
+     * @param context     The application context
+     * @param packageName The package name of the application
+     * @return category + categoryId
+     */
+    static Observable<Pair<String, String>> getPlayStoreAppCategory(final Context context, final String packageName, final String tag) {
+        return Observable.fromEmitter(emitter -> {
+				try {
+					 if (Connectivity.isConnected(context)) {
+
+/*
```
Comment (aritraroy) [inline rxmagneto/src/main/java/com/aritraroy/rxmagneto/RxMagnetoInternal.java:241]: These comments should be removed. A code clean-up is necessary before raising a PR.

File: rxmagneto/src/main/java/com/aritraroy/rxmagneto/RxMagnetoInternal.java [hunk 7]
```diff
@@ -26,8 +29,16 @@
 
 public class RxMagnetoInternal {
 
-    private static final int DEFAULT_TIMEOUT = 5000;
-    static final String MARKET_PLAY_STORE_URL = "https://play.google.com/store/apps/details?id=";
+   private static final int DEFAULT_TIMEOUT = 5000;
+
+	/**
```
Comment (aritraroy) [inline rxmagneto/src/main/java/com/aritraroy/rxmagneto/RxMagnetoInternal.java:34]: Does taking the language into account provide any additional benefits?

**Review by aritraroy — COMMENTED** [review]
Please fix the current issues in the PR.

**Comment by pdalfarr** [comment]
Hi,

I don't have time for this. Just use it if you want ;-)

Thanks

**Comment by aritraroy** [comment]
Okay. Thank you for your contribution but I will have to reject this PR. I liked the features that you have added so, I will add them myself when I have the time. :-)

**Comment by pdalfarr** [comment]
OK, no problem ;-)

I am not used to Github pull requests (never did it) and I really don't
have time to learn how it works (for the moment)

Also, I would suggest to use the 'get all info in one shot' classes I sent
you.

Bye :-)

**Comment by aritraroy** [comment]
I can understand. It is not a problem. I will use your code and make it properly compatible with the current project.
